docs/public/alizer-spec.md

```diff
@@ -0,0 +1,129 @@
+
+- Java library
+- CLI
+- NPM package
+- Go library
+
+so that it can be integrated easily in other projects.
+
+### Inner-loop scenario
+
+As a Vscode/IntelliJ plugin developer, I would like to detect the languages, tools, frameworks, services used within the
+opened user project in the IDE to provide a better user experience and tailor the functionalities I offer them.
+
+## Language Detection
+
+Language detection is based on the file `languages.yml` taken from the [GitHub's Linguist project](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml).
+Because of that, Alizer is able to recognize almost any programming languages, with a customized deeper detection of the
+ones listed below:
+
+- Java
+- Python
+- Javascript
+- Typescript
+
+Language recognition is performed by using files extensions within the source, and only the languages with a significant
+presence (>2% total files) are taken into account for further calculations.
+If Java, Python, Javascript or Typescript languages are detected, Alizer proceeds to check for the presence of tools and 
+frameworks.
+
+### Java
+
+The first step in our deeper Java detection is finding the configuration file used. If a `pom.xml` file is discovered, 
+Alizer assumes is a Maven project. The same for a `build.gradle` file which is a Gradle Project or a `build.xml` for an
+Ant project.
+
+NOTE: Maven, Gradle and Ant are saved as Tools inside the Language
+
+By reading the content of its configuration file and mainly its dependencies section, Alizer is also able to detect 
+frameworks. Currently, it recognizes:
+
+- Micronaut
+- OpenLiberty 
+- Quarkus
+- SpringBoot
+- Vertx
+
+```
+{ 
+    name: 'java', 
+    tools: [ 'maven' ], 
+    frameworks: [ 'quarkus' ] 
+}
+```
+
+### Javascript/Typescript
+
+The detection for Javascript/Typescript works similar to Java. The first thing Alizer does is to check if a `package.json`
+file is in the project. If so, Alizer assumes it is a NodeJs project.
+
+At this point, it reads its content looking for dependencies to discover frameworks. Currently, it recognizes:
+
+- ExpressJs
+- ReactJS
+
+```
+{ 
+    name: 'javascript', 
+    tools: [ 'nodejs' ], 
+    frameworks: [ 'express' ] 
+}
+```
+
+### Python
+
+Currently, the only Python framework Alizer is able to detect is Django.
+To recognize it, it scans all files within the source looking for a file such as `manage.py`, `urls.py`, `wsgi.py`, 
+`asgi.py`. If at least one of them if discovered, it checks its content looking for a django import.
+
+```
+{ 
+    name: 'python', 
+    tools: [], 
+    frameworks: [ 'django' ] 
+}
+```
+
+## DevFile detection
+
+It is also possible to select a devfile from a list of existing devfiles (from a devfile registry or other storage) 
+based on information that Alizer extracts from the source.
+
+If a registry url is fed to Alizer, it will try to download all devfiles from it and select the one which fits best the 
```

Comment:
The general rules is that the detect primitive is accepting a list of devfile meta data which needs to be provided by caller but they can be dowloaded from a registry through another module (registry-support) thus it's only for the CLI. Also we needs to be clear what's available per language so a feature table would be good at the end